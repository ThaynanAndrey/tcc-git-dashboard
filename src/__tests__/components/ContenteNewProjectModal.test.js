import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import ContenteNewProjectModal from '../../components/project/modal/ContentNewProjectModal';

describe('Test ContenteNewProjectModal', () => {
  it('renders correctly a component to show ContenteNewProjectModal', () => {
    const tree = mount(
        <ContenteNewProjectModal />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show ContenteNewProjectModal with correct props', () => {
    const handleChange = jest.fn();

    const tree = mount(
        <ContenteNewProjectModal
            handleChange={handleChange}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('handleChange')).toEqual(handleChange);
  });

  it('renders correctly a component to show ContenteNewProjectModal with correct classes', () => {
    const handleChange = jest.fn();

    const tree = mount(
        <ContenteNewProjectModal
            handleChange={handleChange}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(".modal-content").exists()).toBeTruthy();
  });
});