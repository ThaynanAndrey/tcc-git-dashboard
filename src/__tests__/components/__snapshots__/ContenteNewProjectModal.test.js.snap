// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ContenteNewProjectModal renders correctly a component to show ContenteNewProjectModal 1`] = `
<ContenteNewProjectModal>
  <div
    className="modal-content"
  >
    <h4>
      Novo Projeto
    </h4>
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="input-field col s5"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons prefix"
          >
            insert_drive_file
          </i>
          <input
            name="projectName"
            placeholder="Nome Projeto"
            type="tel"
          />
        </div>
      </form>
    </div>
  </div>
</ContenteNewProjectModal>
`;

exports[`Test ContenteNewProjectModal renders correctly a component to show ContenteNewProjectModal with correct classes 1`] = `
<ContenteNewProjectModal
  handleChange={[MockFunction]}
>
  <div
    className="modal-content"
  >
    <h4>
      Novo Projeto
    </h4>
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="input-field col s5"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons prefix"
          >
            insert_drive_file
          </i>
          <input
            name="projectName"
            onChange={[MockFunction]}
            placeholder="Nome Projeto"
            type="tel"
          />
        </div>
      </form>
    </div>
  </div>
</ContenteNewProjectModal>
`;

exports[`Test ContenteNewProjectModal renders correctly a component to show ContenteNewProjectModal with correct props 1`] = `
<ContenteNewProjectModal
  handleChange={[MockFunction]}
>
  <div
    className="modal-content"
  >
    <h4>
      Novo Projeto
    </h4>
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="input-field col s5"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons prefix"
          >
            insert_drive_file
          </i>
          <input
            name="projectName"
            onChange={[MockFunction]}
            placeholder="Nome Projeto"
            type="tel"
          />
        </div>
      </form>
    </div>
  </div>
</ContenteNewProjectModal>
`;
